### DISCUSSION TITLE
Fabric.js : Text with pattern render very slow in Firefox browser

### DISCUSSION THREAD

**Discussion by niketaOptimumnbrew** [fabricjs/fabric.js#9931] [Q&A]
Hey,

I'm using fabric.js version 5.3.0.
browser : Firefox

I'm adding pattern on text and rendering it on canvas with shadow property.
when I type text that time up to 7-8 line working prefect but above that text render very slow and also browser get's lacking.

I have debug the text case and verify that it only happen when text with pattern have shadow.

Any idea why it's happening ?

**Comment by asturur** [comment]
Yes the patterns in text hit a very slow path. Can you show how your code is generating the text? how the pattern is configured? there are some options that are slower than others

**Reply by niketaOptimumnbrew** [reply]
Thank you @asturur  for answering my question.

Here is my code:

```
options = {
          left:  10,
          top: 10,
          fontSize: 108,
         width : 630,
          textAlign: 'left',
          lockScalingX: false,
          lockScalingY: false,
          hasRotatingPoint: false,
          textBackgroundColor: " ",
          charSpacing: 0,
          centeredRotation: true,
          shadow: {
              "color": '#000000',
              "blur": 5,
              "offsetX": 3,
              "offsetY": 2
              "affectStroke": false
          },
          fontFamily: "BeachTypeMedium",
          opacity: 1,
          strokeUniform: false,
          shadowUniform: true,
          _prevObjectStacking: null,
          statefullCache: true,
          dynamicMinWidth: true,
          editable: true,
      }
       let iTextSample = new fabric.Textbox('SAMPLE TEXT', options);
       fabric.util.loadImage(src, function (img) {
          if (img == null) {
              return;
          }
       iTextSample.set('fill', new fabric.Pattern({
              source: img,
              repeat: 'repeat',
              crossOrigin: 'anonymous',
          }));
       iTextSample.setCoords();
      canvas.add(iTextSample);
```

**Reply by asturur** [reply]
I dont see anything bad, if is slow just under firefox may be a firefox issue. How does it behave in chrome or safari?

**Reply by niketaOptimumnbrew** [reply]
> I dont see anything bad, if is slow just under firefox may be a firefox issue. How does it behave in chrome or safari?

yes in safari and chrome also generated.
